### PR TITLE
Adding namespaceLabels and namespaceAnnotations as TargetCustomization

### PR DESCRIPTION
**Pull request by Tommy12789**
Refers to #2262 #2441 

Added a new parameter namespaceLabels in bundle targets to customize bundleDeployment with those labels

### CODE REVIEW

File: pkg/apis/fleet.cattle.io/v1alpha1/bundle_types.go [hunk 1]
```diff
@@ -221,6 +221,12 @@ type BundleTarget struct {
 	ClusterGroupSelector *metav1.LabelSelector `json:"clusterGroupSelector,omitempty"`
 	// DoNotDeploy if set to true, will not deploy to this target.
 	DoNotDeploy bool `json:"doNotDeploy,omitempty"`
+	// NamespaceLabels are labels that will be appended to the namespace created by Fleet.
+	// +nullable
+	NamespaceLabels *map[string]string `json:"namespaceLabels,omitempty"`
```
Comment (manno) [inline pkg/apis/fleet.cattle.io/v1alpha1/bundle_types.go:226]: I see this is already a pointer to a map in the bundledeployment type. Is that really necessary? Looking at corev1 in k8s, it seems `+optional` and `omitempty` is sufficient, unless we really need to differentiate between a null value and an empty map.

Oh, it seems we do: https://github.com/rancher/fleet/blob/9d2019d87d1a276d79a1cb41edfee1f1e141263a/internal/cmd/agent/deployer/deployer.go#L181-L181
That code deletes all annotations if it's an empty map, but not if it's nil. 👍

File: internal/cmd/controller/target/target.go [hunk 2]
```diff
@@ -58,6 +58,22 @@ func (t *Target) BundleDeployment() *fleet.BundleDeployment {
 		Spec: t.Deployment.Spec,
 	}
 	bd.Spec.Paused = t.IsPaused()
+	for _, bundleTarget := range t.Bundle.Spec.Targets {
+		if bundleTarget.NamespaceLabels != nil {
+			for key, value := range *bundleTarget.NamespaceLabels {
+				(*bd.Spec.Options.NamespaceLabels)[key] = value
```
[the remark this answers is not in the document]
Comment (Tommy12789) [reply internal/cmd/controller/target/target.go:64]: I tried but not working even with initialising the value to a map

[the remark this answers is not in the document]
Comment (manno) [reply internal/cmd/controller/target/target.go:64]: Ah makes sense. I'm at odds with the pointer to the map.

File: pkg/apis/fleet.cattle.io/v1alpha1/bundle_types.go (hunk 1 above)
Comment (manno) [reply pkg/apis/fleet.cattle.io/v1alpha1/bundle_types.go:226]: However that would still work with a map, as a map is effectively a pointer.

**Review by manno — APPROVED** [review]

**Pull request merged**